engine/crates/parser-sdl/src/rules/basic_type.rs

```diff
@@ -42,7 +42,8 @@ impl<'a> Visitor<'a> for BasicType {
         let directives = &type_definition.node.directives;
 
         if ["Query", "Mutation"].contains(&type_definition.node.name.node.as_str())
-            | directives.iter().any(|directive| directive.is_model())
+            || directives.iter().any(|directive| directive.is_model())
+            || type_definition.node.extend
```

Comment:
I think some of my federation directive code was kind of relying on the fact that extensions went through this code path (which is probably what's causing the errors in tests).

Probably need to make sure all the federation directive stuff is present in `ExtendConnectorTypes` as well now

Reply:
Good to know, I'll do some discovery and maybe write tests to make sure that doesn't break.

Reply:
It looks like the federation directives _should_ be handled here: https://github.com/grafbase/grafbase/blob/6a5ec45803b4d698f217936fcc80593f1aadcf4e/engine/crates/parser-sdl/src/rules/extend_connector_types.rs#L47

I'll write at least one test to confirm though.

Reply:
Done in [`7774665` (#1143)](https://github.com/grafbase/grafbase/pull/1143/commits/777466549ddc3be5c31f178430274a886209f0af)

Happy to dig in more if that doesn't build enough confidence.